components/public-api-server/pkg/apiv1/oidc.go

```diff
@@ -350,3 +354,39 @@ func assertIssuerIsReachable(host string) error {
 	}
 	return nil
 }
+
+func assertIssuerProvidesDiscovery(issuer string) error {
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		Proxy:           http.ProxyFromEnvironment,
+	}
+	client := &http.Client{
+		Transport: tr,
+		Timeout:   2 * time.Second,
+		// never follow redirects
+		CheckRedirect: func(*http.Request, []*http.Request) error {
+			return http.ErrUseLastResponse
+		},
+	}
+
+	resp, err := client.Get(issuer + "/.well-known/openid-configuration")
```

Comment:
Can we use this? https://github.com/coreos/go-oidc/blob/921a42b4b63d5b618d17d7e7a36a4acb59e643c6/oidc/oidc.go#L201

Reply:
That's an interesting idea. 

We're not `goidc` instead of `coreos`, but that would work similar. Let me try.

Reply:
It doesn't matter which lib we're using, we're just interested in using the reading of the OpenID Config under well-known endpoint, such that we don't have to implement it entirely ourselves. For that it's even fine if we use two different libraries, given the extra weight is dropped at compile time

Reply:
Just meant, we can use the "official" library the same way. 

I'm following up on the other side at the moment, at the moment, but tempted to create a separate PR to that because for the LOC to be touched. It's one thing to validate the advertised config of an IdP, which is a very sound idea, thanks! There is another valuable change to get rid of the crappy fixtures for the IdP endpoints, replacing them  by real IdP instantiated in tests. As such, we can properly configure examples which are closer to real world.

Doing that on top of this PR.